app/assets/javascripts/discourse/app/lib/user-menu/base-item.js

```diff
@@ -1,7 +1,15 @@
+import UserMenuIconAvatar from "discourse/components/user-menu/icon-avatar";
 import { wantsNewWindow } from "discourse/lib/intercept-click";
+import { withPluginApi } from "discourse/lib/plugin-api";
 import DiscourseURL from "discourse/lib/url";
 
 export default class UserMenuBaseItem {
+  constructor() {
```

Comment:
Since this is just a POJC and not a glimmer component, we aren't able to use `inject` for services. It's tricky because the subclasses of this have constructors that take site, sitesettings and other services as arguments. But I want access here.. It's a big hacky, curious for better suggestions.

Reply:
All the subclasses should be calling `super(...arguments)` in their constructor, so perhaps you could still access them here?

```js
constructor({siteSettings, site}){
  this.siteSettings = siteSettings;
  this.site = site;
}
```